src/Hl7.Fhir.ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -267,12 +279,22 @@ private IEnumerable<TypedElementOnSourceNode> enumerateElements(Dictionary<strin
                 var prettyPath =
                  hit && !info.IsCollection ? $"{ShortPath}.{info.ElementName}" : $"{ShortPath}.{scan.Name}[{_nameIndex}]";
 
-                yield return new TypedElementOnSourceNode(this, scan, info, instanceType, prettyPath);
+                if(!isXhtmlNode) source = scan;
+
+                yield return new TypedElementOnSourceNode(this, source, info, instanceType, prettyPath);
             }
         }
 
         public IEnumerable<ITypedElement> Children(string name = null)
         {
+            //If we have a xhtml typed node and there was not an div tag around the content
+            //the won´t look at the children of this node, since there will be no types
+            //matching the html tags.
+            if (this.InstanceType == "xhtml" && Name != "div")
```

Comment:
This feels like duplicate code, since in FHIR we already had xhtml + div nodes too, and so this case is already handled. Possibly in enumerateElements() (sorry, cannot double check from github web interface where it is exactly).